### PR TITLE
Do not serialize isJoda flag to v8 nodes

### PR DESCRIPTION
**Pull request by pgomulka**
The flag serialised in DocValueFormat.DateTime should only be
serialised/deserialised on [7.7-8.0) nodes (8 excluding).
relates #53477
relates #53478

### CODE REVIEW

**Review by benwtrent — COMMENTED** [review]
I think this is not strictly necessary if v8 handles the bwc problems. See https://github.com/elastic/elasticsearch/pull/53481

File: server/src/main/java/org/elasticsearch/search/DocValueFormat.java [hunk 1]
```diff
@@ -211,8 +211,9 @@ public DateTime(StreamInput in) throws IOException {
                 this.resolution = DateFieldMapper.Resolution.ofOrdinal(in.readVInt());
             }
             final boolean isJoda;
-            if (in.getVersion().onOrAfter(Version.V_7_7_0)) {
-                //if stream is from 7.7 Node it will have a flag indicating if format is joda
+            if (in.getVersion().onOrAfter(Version.V_7_7_0) && in.getVersion().major == 7) {
```
Comment (rjernst) [inline server/src/main/java/org/elasticsearch/search/DocValueFormat.java]: What is this condition trying to accomplish? The major version of 7.7.0 is already 7, so it seems redundant

Comment (rjernst) [reply server/src/main/java/org/elasticsearch/search/DocValueFormat.java]: I see the intent now. I agree with @benwtrent's assessment; backcompat should be handled from the future version, not here.

**Comment by pgomulka** [comment]
apologies, this should have been closed earlier in favour of https://github.com/elastic/elasticsearch/pull/53481

**Pull request closed without merging**